trino/dbapi.py

```diff
@@ -465,7 +465,28 @@ def execute(self, operation, params=None):
         return result
 
     def executemany(self, operation, seq_of_params):
-        raise trino.exceptions.NotSupportedError
+        """
+        PEP-0249: Prepare a database operation (query or command) and then
+        execute it against all parameter sequences or mappings found in the sequence seq_of_parameters.
+        Modules are free to implement this method using multiple calls to
+        the .execute() method or by using array operations to have the
+        database process the sequence as a whole in one call.
+
+        Use of this method for an operation which produces one or more result
+        sets constitutes undefined behavior, and the implementation is permitted (but not required)
+        to raise an exception when it detects that a result set has been created by an invocation of the operation.
+
+        The same comments as for .execute() also apply accordingly to this method.
+
+        Return values are not defined.
+        """
+        for parameters in seq_of_params[:-1]:
+            self.execute(operation, parameters)
+            self.fetchall()
+        if seq_of_params:
+            self.execute(operation, seq_of_params[-1])
+        else:
+            self.execute(operation)
```

Comment:
Why this is needed?

Reply:
I note that the question of what to do when the parameter list is empty has been a question before: https://github.com/winterland1989/mysql-haskell/issues/30

However, SQLlite3 expressly permits executemany with an empy parameter list.  So my presumption would be this implementation is designed to support calling contexts that either strictly require, or do not strictly require, a non-empty parameter list.

Reply:
@hovaesco This line is called when the method was called without parameters. I'm fine to throw an error for starting conservatively.

Reply:
I don't have a strong opinion about doing this in either way. Some DB-API clients supports that and some do not. DB-API spec doesn't say anything about supporting that. 

I'm fine with supporting it here, just please add a test case for it.

Reply:
The test case already exists in test_execute_many_without_params.